app/src/main/java/com/spendster/presentation/addExpenses/chooseCategory/ChooseCategoryActivity.java

```diff
@@ -0,0 +1,54 @@
+package com.spendster.presentation.addExpenses.chooseCategory;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.GridLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.widget.Toast;
+
+import com.google.gson.Gson;
+import com.spendster.R;
+import com.spendster.data.entity.Category;
+
+import java.util.List;
+
+public class ChooseCategoryActivity extends AppCompatActivity implements CategoryView {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_choose_category);
+        ChooseCategoryPresenter chooseCategoryPresenter = new ChooseCategoryPresenter(this,
+                new SimpleCategoryModel());
+        chooseCategoryPresenter.fetchCategories();
+    }
+
+    private void passData(Category item) {
```

Comment:
@SimonMaha poor naming. It's not clear where to pass data exactly from a method name